src/plugin/upload/index.ts

```diff
@@ -0,0 +1,100 @@
+import fs from "fs/promises";
+import { confirm } from "@inquirer/prompts";
+import { PluginZip } from "../core";
+import {
+  buildRestAPIClient,
+  type RestAPIClientOptions,
+} from "../../kintone/client";
+import { logger } from "../../utils/log";
+import os from "os";
+import * as chokidar from "chokidar";
+
+export type Params = {
+  pluginFilePath: string;
+  force?: boolean;
+  watch?: boolean;
+};
+
+export const upload = async (
+  params: Params & RestAPIClientOptions,
+): Promise<void> => {
+  const { pluginFilePath, ...restApiClientOptions } = params;
+
+  const apiClient = buildRestAPIClient(restApiClientOptions);
+
+  // Read plugin zip file
+  const buffer = await fs.readFile(pluginFilePath);
+  const pluginZip = await PluginZip.fromBuffer(buffer);
+  const pluginId = await pluginZip.getPluginID();
+  const pluginManifest = await pluginZip.manifest();
+
+  // Read plugin info from kintone
+  const { plugins: installedPlugins } = await apiClient.plugin.getPlugins(
+    // TODO: rest-api-client should accept array of ids
+    {
+      ids: [pluginId],
+    } as any,
+  );
+  const installedPlugin = installedPlugins.find((p) => p.id === pluginId);
+  const isInstalled = installedPlugin !== undefined;
+
+  const isSameVersion = installedPlugin?.version === pluginManifest.version;
+
+  // Show installation summary
+  const installationSummary = `
+  Installation Summary:
+    Destination: ${restApiClientOptions.baseUrl}
+    File Path: ${pluginFilePath}
+    Plugin ID: ${pluginId}
+    Plugin Name: ${pluginManifest.name}
+    Current version: ${installedPlugin?.version ?? "(not installed)"}
+    Target version: ${pluginManifest.version}${isSameVersion ? " (reinstall)" : ""}`;
+  logger.info(installationSummary);
+
+  // Get confirmation from user if required
+  if (!params.force && !params.watch) {
+    const answers = await confirm({
+      message: `Do you continue to add this plugin?`,
```

Comment:
英語の文法的に怪しい気がする
```suggestion
      message: `Do you want to continue adding this plugin?`,
```
もしくは単に
```suggestion
      message: `Do you want to continue?`,
```
(大体のインストーラーは後者のイメージ)

Reply:
実行時点でアップロードの意図は明確だし後者で良さそう